packages/gateway/src/routes/verifyCode/handler.ts

```diff
@@ -0,0 +1,251 @@
+}
+
+export async function deleteUsedVerificationCode(
+  nonce: string
+): Promise<boolean> {
+  const count = await del(`verification_${nonce}`)
+  return Boolean(count)
+}
+
+export function generateNonce() {
+  return crypto.randomBytes(16).toString('base64').toString()
+}
+
+export async function sendVerificationCode(
+  mobile: string,
+  verificationCode: string
+): Promise<void> {
+  const params = {
+    msisdn: mobile,
+    code: verificationCode
+  }
+
+  await fetch(resolve(NOTIFICATION_URL, 'authenticationCode'), {
+    method: 'POST',
+    body: JSON.stringify(params),
+    headers: {
+      Authorization: `Bearer ${await createToken(
+        'auth',
+        ['service'],
+        ['opencrvs:notification-user'],
+        JWT_ISSUER
+      )}`
+    }
+  })
+
+  return undefined
+}
+
+export async function createToken(
+  userId: string,
+  scope: string[],
+  audience: string[],
+  issuer: string,
+  system?: boolean
+): Promise<string> {
+  if (typeof userId === undefined) {
+    throw new Error('Invalid userId found for token creation')
+  }
+  return sign({ scope }, cert, {
+    subject: userId,
+    algorithm: 'RS256',
+    expiresIn: system
+      ? CONFIG_SYSTEM_TOKEN_EXPIRY_SECONDS
+      : CONFIG_TOKEN_EXPIRY_SECONDS,
+    audience,
+    issuer
+  })
+}
+
+export async function generateAndSendVerificationCode(
+  nonce: string,
+  mobile: string,
+  scope: string[]
+) {
+  const isDemoUser = scope.indexOf('demo') > -1
+  logger.info('isDemoUser', {
+    isDemoUser: isDemoUser
+  })
+  let verificationCode
+  if (isDemoUser) {
+    verificationCode = '000000'
+    await storeVerificationCode(nonce, verificationCode)
+  } else {
+    verificationCode = await generateVerificationCode(nonce, mobile)
+  }
+  if (!PRODUCTION || QA_ENV) {
+    logger.info('Sending a verification SMS', {
+      mobile: mobile,
+      verificationCode
+    })
+  } else {
+    if (isDemoUser) {
+      throw unauthorized()
+    } else {
+      await sendVerificationCode(mobile, verificationCode)
+    }
+  }
+}
+
+export default async function sendVerifyCodeHandler(
+  request: Hapi.Request,
+  h: Hapi.ResponseToolkit
+): Promise<ISendVerifyCodeResponse> {
+  const token = request.headers.authorization.replace('Bearer ', '') as string
+  const tokenDecoded = verifyToken(token)
+  const payload = request.payload as ISendVerifyCodePayload
+  const { phoneNumber } = payload
+  const { sub: userId, scope } = tokenDecoded
+
+  const nonce = generateNonce()
+  const response: ISendVerifyCodeResponse = {
+    userId,
+    nonce,
+    mobile: phoneNumber,
+    status: 'Success'
+  }
+  await generateAndSendVerificationCode(nonce, phoneNumber, scope)
+  return response
+}
+
+export const requestSchema = Joi.object({
+  phoneNumber: Joi.string()
+})
+
+export const responseSchema = Joi.object({
+  userId: Joi.string(),
+  nonce: Joi.string(),
+  mobile: Joi.string(),
+  status: Joi.string()
+})
+
+/* tslint:disable */
```

Comment:
Maybe in this case it would be better to rename the variable as suggested. I see you've done it similarly as how we've done it in `packages/auth/src/features/authenticate/service.ts`, but as it's such a small change, I'd remove the tslint:disable in both files and rename the variables